### PR TITLE
Extract List Offline Region and Download Region dialogs in Fragments

### PR DESCRIPTION
**Pull request by AlexisTabin**
- Extract List Offline Region and Download Region dialogs in Fragments

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/ui/maps/offline/OfflineManagerActivity.kt [hunk 1]
```diff
@@ -81,41 +81,15 @@ class OfflineManagerActivity : MapViewBaseActivity(), OnMapReadyCallback {
...
+    fun downloadRegionDialog(v: View) {
+        DownloadRegionDialogFragment().show(supportFragmentManager, this.getString(R.string.download_region_dialog_fragment))
     }
 
     /**
      * Define offline region parameters, including bounds,
      * min/max zoom, and metadata
      */
-    private fun downloadRegion(regionName: String) {
+    fun downloadRegion(regionName: String) {
```
Comment (Sporarum) [inline app/src/main/java/ch/epfl/sdp/ui/maps/offline/OfflineManagerActivity.kt:92]: Does this actually download a region ?
If yes, the javadoc needs to be updated
If not, it should be retitled

Comment (AlexisTabin) [reply app/src/main/java/ch/epfl/sdp/ui/maps/offline/OfflineManagerActivity.kt:96]: It prepares the download, I updated the javadoc

File: app/src/main/java/ch/epfl/sdp/ui/maps/offline/ListOfflineRegionDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.offline.OfflineRegion
+
+class ListOfflineRegionDialogFragment(private val items: Array<CharSequence>,
+                                      private val offlineRegions: Array<OfflineRegion>,
+                                      private val mapboxMap: MapboxMap,
+                                      private val progressBar: ProgressBar,
+                                      private val mapView: MapView) : DialogFragment() {
+    private var regionSelected = 0
```
Comment (Ph0tonic) [inline app/src/main/java/ch/epfl/sdp/ui/maps/offline/ListOfflineRegionDialogFragment.kt:22]: Could not we move this variable inside onCreateDialog method ?

Comment (AlexisTabin) [reply app/src/main/java/ch/epfl/sdp/ui/maps/offline/ListOfflineRegionDialogFragment.kt:22]: Moved :)

**Pull request merged**